cassandra/common/src/main/java/org/opennms/newts/cassandra/CassandraSessionImpl.java

```diff
@@ -15,115 +15,144 @@
 
-        LOG.info("Setting up session with {}:{} using compression {}", hostname, port, compression.toUpperCase());
+        LOG.info("Setting up session with to {}:{} using compression {} and local datacenter: {}", hostname, port,
+                compression.toUpperCase(), localDatacenter);
+
+        ProgrammaticDriverConfigLoaderBuilder configBuilder = DriverConfigLoader.programmaticBuilder()
+                .startProfile("default");
+        configBuilder.withString(DefaultDriverOption.SESSION_KEYSPACE, keyspace);
+        configBuilder.withStringList(DefaultDriverOption.CONTACT_POINTS, toContactPoints(hostname, port));
 
-        final PoolingOptions poolingOptions = new PoolingOptions();
         if (coreConnectionsPerHost != null) {
-            LOG.debug("Using {} core connections per host.", coreConnectionsPerHost);
-            poolingOptions.setCoreConnectionsPerHost(HostDistance.LOCAL, coreConnectionsPerHost)
-                    .setCoreConnectionsPerHost(HostDistance.REMOTE, coreConnectionsPerHost);
+            LOG.warn("Ignoring core connections per host value of: {}. " +
+                    "Use 'max-connections-per-host' only instead.", coreConnectionsPerHost);
+
         }
         if (maxConnectionsPerHost != null) {
             LOG.debug("Using {} max connections per host.", maxConnectionsPerHost);
-            poolingOptions.setMaxConnectionsPerHost(HostDistance.LOCAL, maxConnectionsPerHost)
-                    .setMaxConnectionsPerHost(HostDistance.REMOTE, maxConnectionsPerHost);
+            configBuilder.withInt(DefaultDriverOption.CONNECTION_POOL_LOCAL_SIZE, maxConnectionsPerHost);
+            configBuilder.withInt(DefaultDriverOption.CONNECTION_POOL_REMOTE_SIZE, maxConnectionsPerHost);
         }
         if (maxRequestsPerConnection != null) {
             LOG.debug("Using {} max requests per connection.", maxRequestsPerConnection);
-            poolingOptions.setMaxRequestsPerConnection(HostDistance.LOCAL, maxRequestsPerConnection)
-                    .setMaxRequestsPerConnection(HostDistance.REMOTE, maxRequestsPerConnection);
+            configBuilder.withInt(DefaultDriverOption.CONNECTION_MAX_REQUESTS, maxRequestsPerConnection);
         }
-
-        Builder builder = Cluster
-                .builder()
-                .withPort(port)
-                .addContactPoints(hostname.split(","))
-                .withReconnectionPolicy(new ExponentialReconnectionPolicy(1000, 2 * 60 * 1000))
-                .withPoolingOptions(poolingOptions)
-                .withCompression(Compression.valueOf(compression.toUpperCase()));
+        configBuilder.withClass(DefaultDriverOption.RECONNECTION_POLICY_CLASS, ExponentialReconnectionPolicy.class);
+        configBuilder.withDuration(DefaultDriverOption.RECONNECTION_BASE_DELAY, Duration.of(1, ChronoUnit.SECONDS));
+        configBuilder.withDuration(DefaultDriverOption.RECONNECTION_MAX_DELAY, Duration.of(2, ChronoUnit.MINUTES));
```

Comment:
Is this something we could conceivably want to make configurable?

Reply:
Possibly yes. This and all the other settings can be tuned when the "driver settings file" is used - will add some docs to that effect in OpenNMS.